### ISSUE TITLE
Cannot delete products from compare list

### ISSUE THREAD

**Issue by SLiCK-303** [thirtybees/thirtybees#791]
i add 3 products to compare
then i delete 2 products and they dissapear
when i click to delete 3rd product, all 3 products reapear

Same happens if i compare just 2 products. When i want to delete last one, they magically reapear.

https://forum.thirtybees.com/topic/2331/cannot-delete-products-from-compare-list

**Comment by getdatakick** [comment]
I could not reproduce this problem. @SLiCK-303, is this still reproducible on 1.0.8?

**Comment by SLiCK-303** [comment]
yup, just checked it again.  Add 2 products to the compare list, then compare them.  Then delete them on that screen one at a time.  You will find that when you delete the last one, both will show up again.

**Comment by getdatakick** [comment]
Unfortunately I can't reproduce it on my vanilla installation 1.0.8

**Comment by SLiCK-303** [comment]
try it on a upgraded to 1.0.8 setup...

**Comment by NemoPS** [comment]
It could be related to something else if it's only on the upgraded version, working on a vanilla 1.0.8 and on the latest github build, closing for now.

**Comment by Traumflug** [comment]
I can reproduce this on latest _1.0.x_ (using Firefox).

[screencast.zip](https://github.com/thirtybees/thirtybees/files/2999573/screencast.zip)

Another observation, probably related:

- Click a category, e.g. _Coffee and Tea_.
- Add three products to the comparison.
- Click _Compare (3)_.
- Click cross in a product column to remove one product.
- Click on the same category in the menu bar.
- ==> Compare button reads again _Compare (3)_, not _Compare (2)_. Clicking it also shows all three products again.

**Comment by getdatakick** [comment]
Reproduced on firefox. 

Function reloadProductComparison from product-comparison.js triggers ajax, and then immediately changes window location, which triggers page reload. Firefox uses queuing mechanism, reloading page effectively blocks ajax call from running.
